sdk/openai/Azure.AI.OpenAI/src/Generated/AudioTranscriptionOptions.Serialization.cs

```diff
@@ -193,6 +201,59 @@ internal static AudioTranscriptionOptions DeserializeAudioTranscriptionOptions(J
                 serializedAdditionalRawData);
         }
 
+        private BinaryData SerializeMultipart(ModelReaderWriterOptions options)
+        {
+            using MultipartFormDataRequestContent content = ToMultipartRequestContent();
+            using MemoryStream stream = new MemoryStream();
+            content.WriteTo(stream);
+            if (stream.Position > int.MaxValue)
+            {
+                return BinaryData.FromStream(stream);
+            }
+            else
+            {
+                return new BinaryData(stream.GetBuffer().AsMemory(0, (int)stream.Position));
+            }
+        }
+
+        internal virtual MultipartFormDataRequestContent ToMultipartRequestContent()
+        {
+            MultipartFormDataRequestContent content = new MultipartFormDataRequestContent();
+            content.Add(AudioData, "file", "file");
+            if (Optional.IsDefined(Filename))
+            {
+                content.Add(Filename, "filename");
```

Comment:
I believe we should not be writing the filename as a separate property -- at least that is the way it is in the third-party service: https://platform.openai.com/docs/api-reference/audio/createTranscription.  Is the Azure API different here?

Reply:
As above comment, we cannot figure out that the added `Filename` property is a File meta-data, not regular property. It is not correct to add a separate property `FileName` in typespec, instead it should append as decorator of `bytes` when it is interpreted as a file.
Current, we only can customize it.

Reply:
My understanding of the [TypeSpec for multipart/form-data](https://github.com/microsoft/typespec/issues/3046) is that filename is metadata on a file, as shown in this example:
 
![image](https://github.com/Azure/azure-sdk-for-net/assets/21269347/692a17d6-86a7-4798-ad26-a44f7dfc4551)

As such, I believe filename can be associated with the file.  The file name header is not required for multipart parts -- if it is not present, we should not pass a value for the filename property.

In any case, we should not add Filename as a separate content part, since this will cause the service to return an error response.

Reply:
Yes. If the `Filename` is a meta-data of a File we will not add Filename as a separate content part in multipart.
but in OpenAI, the `FileName` is a  property of the model. That's why codegen will send it as a content part. 

Typespec current lack of the capability to define a filename as meta-data of an File.